src/kaleidoscope/hardware/keyboardio/Model01.h

```diff
@@ -25,100 +25,49 @@
 #include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
 #include "KeyboardioScanner.h"
 
-#include "kaleidoscope/macro_helpers.h"
-
 #define CRGB(r,g,b) (cRGB){b, g, r}
 
-#include "kaleidoscope/Hardware.h"
-#include "kaleidoscope/MatrixAddr.h"
+#include "kaleidoscope/hardware/keyboardio/model01/Model01KeyScannerDescription.h"
+#include "kaleidoscope/hardware/keyboardio/model01/Model01KeyScanner.h"
+#include "kaleidoscope/hardware/keyboardio/model01/Model01LedDriver.h"
+#include "kaleidoscope/driver/bootloader/avr/Caterina.h"
+#include "kaleidoscope/hardware/avr/AVRDeviceDescription.h"
+#include "kaleidoscope/hardware/avr/AVRDevice.h"
 
 namespace kaleidoscope {
 namespace hardware {
 namespace keyboardio {
 
-class Model01 : public kaleidoscope::Hardware {
- public:
-  Model01(void);
-
-  static constexpr byte matrix_rows = 4;
-  static constexpr byte matrix_columns = 16;
-  static constexpr int8_t led_count = 64;
+struct Model01DeviceDescription : kaleidoscope::hardware::avr::AVRDeviceDescription {
+  typedef Model01KeyScannerDescription KeyScannerDescription;
+  typedef Model01KeyScanner KeyScanner;
+  typedef Model01LedDriverDescription LEDsDescription;
+  typedef Model01LedDriver LEDs;
+  typedef kaleidoscope::driver::bootloader::avr::Caterina BootLoader;
+};
 
-  typedef MatrixAddr<matrix_rows, matrix_columns> KeyAddr;
-  static constexpr int8_t numKeys() {
-    return matrix_columns * matrix_rows;
-  }
+class Model01 : public kaleidoscope::hardware::avr::AVRDevice<Model01DeviceDescription> {
+ public:
+  static void setup();
 
-  void syncLeds(void);
-  void setCrgbAt(KeyAddr key_addr, cRGB color);
-  DEPRECATED(ROW_COL_FUNC) void setCrgbAt(byte row, byte col, cRGB color) {
-    setCrgbAt(KeyAddr(row, col), color);
-  }
-  void setCrgbAt(int8_t i, cRGB crgb);
-  cRGB getCrgbAt(int8_t i);
-  int8_t getLedIndex(KeyAddr key_addr);
-  DEPRECATED(ROW_COL_FUNC) int8_t getLedIndex(byte row, byte col) {
-    return getLedIndex(KeyAddr(row, col));
-  }
+  static void enableHardwareTestMode();
 
-  void scanMatrix(void);
-  void readMatrix(void);
-  void actOnMatrixScan(void);
-  void setup();
-  void rebootBootloader();
-
-  /* These public functions are things supported by the Model 01, but
-   * aren't necessarily part of the Kaleidoscope API
-   */
-  void enableHighPowerLeds(void);
-  void enableScannerPower(void);
-  void setKeyscanInterval(uint8_t interval);
-  boolean ledPowerFault(void);
-
-  void maskKey(KeyAddr key_addr);
-  DEPRECATED(ROW_COL_FUNC) void maskKey(byte row, byte col) {
-    maskKey(KeyAddr(row, col));
+  void setKeyscanInterval(uint8_t interval) {
+    key_scanner_.setKeyscanInterval(interval);
   }
-  void unMaskKey(KeyAddr key_addr);
-  DEPRECATED(ROW_COL_FUNC) void unMaskKey(byte row, byte col) {
-    unMaskKey(KeyAddr(row, col));
+  keydata_t leftHandState() {
```

Comment:
I believe the only thing that still uses this API is the old Model01 test mode. I'd rather turn that plugin into a no-op with a warning that it's been replaced with the more generic test mode and not expose these methods in the redo.

Reply:
Ok, will do that! Do you want a separate PR for the old Model01 test mode NoOp?

Reply:
Separate commit is good enough